### PR TITLE
update components to use new db

### PR DESCRIPTION
**Pull request by StianHaugland1**

### CODE REVIEW

File: common/utils/string.utils.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import replaceSpecialCharacters from "replace-special-characters";
 
-export const labelToUrlComponent = (label: string): string => {
+export const labelToUrlComponent = (label?: string): string | undefined => {
```
Comment (boyum) [inline common/utils/string.utils.ts:3]: Why add this complexity here and not in the functions which use this function?

File: common/utils/router.utils.ts [hunk 2]
```diff
@@ -29,3 +29,25 @@ export const getLanguagePath = (
...
+  const topicPath = labelToUrlComponent(topic?.translations.get(language.code)?.labels.at(0)?.label)
+  if(!topicPath) return `/${language.code}${search}`
+  const subTopicPath = labelToUrlComponent(subTopic?.translations.get(language.code)?.labels.at(0)?.label)
+  if(!subTopicPath) return `/${language.code}/${topicPath}${search}`;
+  return `/${language.code}/${topicPath}/${subTopicPath}${search}`
+};
+
+export const uriComponentToTopicPath = (uriComponent?: string): string => {
+  if (!uriComponent) return "";
+  return labelToUrlComponent(decodeURIComponent(uriComponent))
```
Comment (henriettemoe) [inline common/utils/router.utils.ts:52]: Receiving a type error here
Skjermbilde 2024-03-06 kl  13 44 55

File: common/types/types.ts [hunk 3]
```diff
@@ -111,10 +111,56 @@ export type Translation = {
...
   lang: Language;
   labels: Labels;
   audioFiles?: Array<AudioFile>;
 };
 
 export type Translations = Record<LanguageCode, Record<string, Translation>>;
+
+
+
+export type NewTranslation = {
```
[the remark this answers is not in the document]
Comment (henriettemoe) [reply common/types/types.ts:124]: Will have to change a lot in the code now if we are to do so. So, we can discuss it.

File: common/utils/string.utils.ts (hunk 1 above)
Comment (StianHaugland1) [reply common/utils/string.utils.ts]: Fixed :D

File: common/utils/searchResults.utils.ts [hunk 4]
```diff
@@ -43,11 +65,27 @@ export const sortSearchByPosition = (
   search: string,
   results: SearchResult[],
 ): SearchResult[] => {
-  const res = sortSearchByTopic(results);
-  return res.toSorted((a, b) => {
+  // const res = sortSearchByTopic(results);
```
Comment (henriettemoe) [inline common/utils/searchResults.utils.ts:68]: Can this be removed or will we need it later?

File: h5p-bildetema/src/components/App/App.tsx [hunk 5]
```diff
@@ -11,16 +12,21 @@ type appProps = {
 };
 
 export const App: FC<appProps> = ({ defaultLanguages, backendUrl }) => {
-  const { isLoading: isLoadingData, data } = useQuery(["dataFromDB"], () =>
-    getData(backendUrl),
-  );
+  const { isLoading, data: newData } = useQuery(["newData"], async () => {
+    return getNewData(backendUrl);
+    // return getNewData("http://127.0.0.1:10000/devstoreaccount1/data/newData.json.tar.gz");
```
Comment (henriettemoe) [inline h5p-bildetema/src/components/App/App.tsx:17]: Does this comment need to be there? If so, comment why

File: h5p-bildetema/src/components/Bildetema/Bildetema.tsx [hunk 6]
```diff
@@ -13,19 +14,32 @@ import { TopicRouteController } from "../TopicRouteController/TopicRouteControll
...
 
 type BildetemaProps = {
   defaultLanguages: string[];
   isLoadingData: boolean;
 };
 
-const staticPaths = ["/sok"];
+const STATIC_PATHS = {
+  SEARCH: "/sok",
+  CUSTOM_VIEW: "/customview",
```
Comment (henriettemoe) [inline h5p-bildetema/src/components/Bildetema/Bildetema.tsx:30]: If we use "customview" (for simplification?) maybe "sok" should change to "search"?

File: h5p-bildetema/src/components/CustomViewPage/CustomViewPage.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+// const SearchParamKeys = {
+//   SEARCH: "search",
+//   FILTER: "filter",
+//   VIEW_LANG: "viewLang",
+//   SEARCH_LANG: "lang",
+//   WORDS: "words",
```
Comment (henriettemoe) [inline h5p-bildetema/src/components/CustomViewPage/CustomViewPage.tsx:6]: Is this something we need to keep?

File: h5p-bildetema/src/components/PrintWords/PrintWords.tsx [hunk 8]
```diff
@@ -98,16 +49,20 @@ export const PrintWords = forwardRef<HTMLDivElement, PrintWordsProps>(
...
-            const activeImage = word.images?.find(image => {
-              return document.querySelector(
-                `div.swiper-slide-active img[src="${image?.src}"]`,
-              );
-            });
+            // const activeImage = word.images?.find(image => {
+            //   return document.querySelector(
+            //     `div.swiper-slide-active img[src="${image?.src}"]`,
+            //   );
+            // });
```
Comment (henriettemoe) [inline h5p-bildetema/src/components/PrintWords/PrintWords.tsx:63]: Why is this removed? Didn't we use it to print the correct image that is shown, so that users can select the preferred image to print?

File: h5p-bildetema/src/main.tsx [hunk 9]
```diff
@@ -38,7 +38,8 @@ ReactDOM.createRoot(document.getElementById("root")!).render(
             <ContentIdContext.Provider value="content-id">
               <App
                 defaultLanguages={["nob"]}
-                backendUrl="https://cdn-prod-bildetema.azureedge.net/data/database.json.tar.gz"
+                backendUrl="https://cdn-dev-bildetema.azureedge.net/data/data.json.tar.gz"
+                // backendUrl="https://cdn-prod-bildetema.azureedge.net/data/database.json.tar.gz"
```
Comment (henriettemoe) [inline h5p-bildetema/src/main.tsx:42]: Should the old backend url still be here? Comment if so, else we can store it somewhere else

File: h5p-editor-bildetema-words-topic-image/src/App.tsx [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import { LanguageCode } from "common/types/LanguageCode";
+// import { LanguageCode } from "common/types/LanguageCode";
```
Comment (henriettemoe) [inline h5p-editor-bildetema-words-topic-image/src/App.tsx:1]: Remove old import if we don't need it

**Comment by henriettemoe** [comment]
Remember to update patch versions in library

File: h5p-bildetema/src/components/Bildetema/Bildetema.tsx (hunk 6 above)
Comment (StianHaugland1) [reply h5p-bildetema/src/components/Bildetema/Bildetema.tsx:30]: agree, can fix this in a later pr

File: h5p-bildetema/src/components/PrintWords/PrintWords.tsx (hunk 8 above)
Comment (StianHaugland1) [reply h5p-bildetema/src/components/PrintWords/PrintWords.tsx]: Fixed

File: common/utils/searchResults.utils.ts (hunk 4 above)
Comment (StianHaugland1) [reply common/utils/searchResults.utils.ts:68]: removed

File: h5p-bildetema/src/components/CustomViewPage/CustomViewPage.tsx (hunk 7 above)
Comment (StianHaugland1) [reply h5p-bildetema/src/components/CustomViewPage/CustomViewPage.tsx:6]: Can be removed for now

**Comment by StianHaugland1** [comment]
We should later rename "newWord, newData" and so on

File: common/types/types.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (StianHaugland1) [reply common/types/types.ts:122]: Should be fixed in a later pr

**Review by henriettemoe — APPROVED** [review]
